core/src/main/java/org/mskcc/oncotree/utils/TumorTypesUtil.java

```diff
@@ -88,15 +90,58 @@ public static Properties getProperties() {
         return filtered;
     }
 
-    public static InputStream getTumorTypeInputStream() {
-        Properties properties = getProperties();
-        InputStream inputStream = null;
+    public static InputStream getTumorTypeInputStream(Map<String, TumorType> tumorTypes) {
+        String tsvAsString = getTsvFromTumorTypes(tumorTypes);
         try {
-            inputStream = new FileInputStream(properties.getProperty("tumor_type_file_path"));
-        } catch (FileNotFoundException e) {
-            e.printStackTrace();
+            return new ByteArrayInputStream(tsvAsString.getBytes("UTF-8"));
+        } catch (UnsupportedEncodingException e) {
```

Comment:
We still need this catch block. The getBytes method can throw an UnsupportedEncodingException.